log/version.go

```diff
@@ -0,0 +1,86 @@
+package log
+
+import (
+	"runtime/debug"
+)
+
+// Version is the version tag value to be added to the global logger.
+//
+// If it's changed to non-empty value before the calling of Init* functions
+// (InitFromConfig/InitLogger/InitLoggerJSON/InitLoggerWithConfig/InitSentry),
+// the global logger initialized will come with a tag of VersionKey("v"),
+// added to every line of logs.
+// For InitSentry the global sentry will also be initialized with a tag of
+// "version".
+//
+// In order to use it, either set it in your main function early,
+// before calling InitLogger* functions,
+// with the value coming from flag/config file/etc..
+// For example:
+//
+//     func main() {
+//       log.Version = *flagVersion
+//       log.InitLoggerJSON(log.Level(*logLevel))
+//       // ...
+//     }
+//
+// Or just "stamp" it during build time,
+// by passing additional ldflags to go build command.
+// For example:
+//
+//     go build -ldflags "-X github.com/reddit/baseplate.go/log.Version=$(git rev-parse HEAD)"
+//
+// Change its value after calling Init* functions will have no effects,
+// unless you call Init* functions again.
+//
+// Starting from go 1.18, when Version is empty string,
+// Init* functions will auto fallback to runtime/debug.ReadBuildInfo,
+// and use that value as the version in logs and sentry reports if available.
+var (
+	VersionLogKey = "v"
+
+	Version string
+)
+
+func getVersion() string {
+	if Version != "" {
+		return Version
+	}
+	info, ok := debug.ReadBuildInfo()
+	if !ok || info == nil {
+		return ""
+	}
+	return getVersionFromBuildInfo(info)
+}
```

Comment:
I haven't cehcked the final implementation, but I think initially it seemed like ReadBuildInfo wasn't something we would want to call often.

How about populating Version in init() based on ReadBuildInfo if it's not already populated?

Reply:
`init()` happens after execve and before calling main(), so it's at runtime -- ldflags happens at compile time.

So we don't need sync.once

Reply:
ok. changed to `init()`.